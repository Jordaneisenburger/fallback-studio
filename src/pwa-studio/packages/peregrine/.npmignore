.eslintrc.js
/lib/**/__{docs,helpers,mocks,tests}__/**
/scripts
