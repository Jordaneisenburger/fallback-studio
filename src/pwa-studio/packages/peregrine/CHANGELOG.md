# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.5.0"></a>
# 0.5.0 (2018-08-13)


### Features

* multicasting REST API client ([#164](https://github.com/magento/pwa-studio/issues/164)) ([2852e14](https://github.com/magento/pwa-studio/commit/2852e14)), closes [#140](https://github.com/magento/pwa-studio/issues/140)
* Simulators for async testing ([#24](https://github.com/magento/pwa-studio/issues/24)) ([a380da9](https://github.com/magento/pwa-studio/commit/a380da9))
