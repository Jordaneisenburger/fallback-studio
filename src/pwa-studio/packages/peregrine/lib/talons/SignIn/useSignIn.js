import { useCallback, useRef } from 'react';
import { useUserContext } from '../../context/user';

// Note: we can't access the actual message that comes back from the server
// without doing some fragile string manipulation. Hardcoded for now.
const ERROR_MESSAGE =
    'The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later.';

export const useSignIn = props => {
    const { setDefaultUsername, showCreateAccount, showForgotPassword } = props;

    const [
        { isGettingDetails, isSigningIn, signInError, getDetailsError },
        { signIn }
    ] = useUserContext();

    const hasError = !!signInError || !!getDetailsError;

    const formRef = useRef(null);
    const errorMessage = hasError ? ERROR_MESSAGE : null;

    const handleSubmit = useCallback(
        ({ email: username, password }) => {
            signIn({ username, password });
        },
        [signIn]
    );

    const handleForgotPassword = useCallback(() => {
        const { current: form } = formRef;

        if (form) {
            setDefaultUsername(form.formApi.getValue('email'));
        }

        showForgotPassword();
    }, [setDefaultUsername, showForgotPassword]);

    const handleCreateAccount = useCallback(() => {
        const { current: form } = formRef;

        if (form) {
            setDefaultUsername(form.formApi.getValue('email'));
        }

        showCreateAccount();
    }, [setDefaultUsername, showCreateAccount]);

    return {
        errorMessage,
        formRef,
        handleCreateAccount,
        handleForgotPassword,
        handleSubmit,
        isBusy: isGettingDetails || isSigningIn
    };
};
