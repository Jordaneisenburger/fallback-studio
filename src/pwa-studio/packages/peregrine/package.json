{
  "name": "@magento/peregrine",
  "version": "2.1.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "The core runtime of Magento PWA",
  "main": "dist/index.js",
  "scripts": {
    "build": "concurrently --raw yarn:build:cjs yarn:build:esm",
    "build:cjs": "BABEL_ENV=production babel src --out-dir dist --root-mode 'upward' --source-maps",
    "build:dev": "yarn run build",
    "build:esm": "BABEL_ENV=development babel src --out-dir esm --root-mode 'upward' --source-maps",
    "clean": "rimraf dist esm",
    "prepublishOnly": "yarn run clean && yarn run build",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "storybook:build": "build-storybook -c .storybook -o storybook-dist",
    "watch": "yarn run build --watch"
  },
  "repository": "github:magento-research/pwa-studio",
  "author": "Magento Commerce",
  "license": "(OSL-3.0 OR AFL-3.0)",
  "bugs": {
    "url": "https://github.com/magento-research/pwa-studio/issues"
  },
  "homepage": "https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine#readme",
  "devDependencies": {
    "@babel/cli": "~7.2.3",
    "@babel/core": "~7.3.4",
    "@babel/plugin-proposal-class-properties": "~7.3.4",
    "@babel/plugin-proposal-object-rest-spread": "~7.3.4",
    "@babel/plugin-syntax-dynamic-import": "~7.2.0",
    "@babel/plugin-syntax-jsx": "~7.2.0",
    "@babel/plugin-transform-react-jsx": "~7.3.0",
    "@babel/plugin-transform-runtime": "~7.4.4",
    "@babel/runtime": "~7.4.2",
    "@storybook/react": "~4.1.13",
    "concurrently": "~4.1.0",
    "intl": "~1.2.5",
    "react": "~16.8.3",
    "react-apollo": "~2.5.3",
    "react-dom": "~16.8.3",
    "react-router-dom": "~5.0.0",
    "react-test-renderer": "~16.8.6",
    "rimraf": "~2.6.3"
  },
  "peerDependencies": {
    "@babel/runtime": "^7.4.2",
    "react": "^16.8.0",
    "react-apollo": "^2.5.3",
    "react-dom": "^16.8.0",
    "react-router-dom": "~5.0.0"
  },
  "engines": {
    "node": "8.10.0 || >=10.7.0",
    "yarn": ">=1.12.0"
  },
  "module": "esm/index.js",
  "jsnext:main": "src/index.js",
  "es2015": "src/index.js",
  "sideEffects": false
}
