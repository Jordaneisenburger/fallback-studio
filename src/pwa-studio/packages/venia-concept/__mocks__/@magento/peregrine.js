import React from 'react';
import { BrowserPersistence } from './util/simplePersistence';

const peregrine = jest.requireActual('../../../peregrine/lib');

// re-exports
const RestApi = {
    Magento2: {
        request: jest.fn()
    }
};

const Util = { BrowserPersistence };

// hooks
const useApolloContext = jest.fn(peregrine.useApolloContext);
const useEventListener = jest.fn(peregrine.useEventListener);
const useDropdown = jest.fn(peregrine.useDropdown);
const useQuery = jest.fn(peregrine.useQuery);
const useQueryResult = jest.fn(peregrine.useQueryResult);
const useSearchParam = jest.fn(peregrine.useSearchParam);
const useWindowSize = jest.fn(peregrine.useWindowSize);
const useToasts = jest.fn(peregrine.useToasts);
// components

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
const Price = jest.fn().mockReturnValue(<div />);

module.exports = {
    ...peregrine,
    Price,
    RestApi,
    Util,
    useApolloContext,
    useEventListener,
    useDropdown,
    useQuery,
    useQueryResult,
    useSearchParam,
    useWindowSize,

    useToasts
};
