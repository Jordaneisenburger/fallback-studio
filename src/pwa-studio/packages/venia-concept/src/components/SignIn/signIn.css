.root {
    display: grid;
    gap: 0.5rem;
    justify-items: stretch;
    padding: 0 1.5rem;
}

.forgotPassword {
    display: block;
    text-decoration: underline;
    color: rgb(var(--venia-error));
    font-size: 0.8rem;
}

.form {
    display: grid;
    row-gap: 1rem;
}

.modal {
    visibility: hidden;
    height: 100%;
    width: 100%;
    background-color: rgb(var(--venia-grey));
    text-align: center;
    position: absolute;
    bottom: 0;
}

.modal_active {
    visibility: visible;
    composes: modal;
    opacity: 0.9;
}

.showCreateAccountButton,
.signInButton {
    text-align: center;
    margin-top: 0.5rem;
}

.signInDivider {
    border-color: rgb(var(--venia-border));
    border-style: solid;
    border-width: 0 0 1px;
    margin: 2rem 1rem;
}

.signInError {
    color: rgb(var(--venia-error));
    text-align: center;
}
