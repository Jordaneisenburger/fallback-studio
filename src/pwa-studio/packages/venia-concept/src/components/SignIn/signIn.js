import React, { Component } from 'react';
import { bool, func, object, shape, string } from 'prop-types';
import { Form } from 'informed';

import Button from 'src/components/Button';
import Field from 'src/components/Field';
import LoadingIndicator from 'src/components/LoadingIndicator';
import TextInput from 'src/components/TextInput';

import { isRequired } from 'src/util/formValidators';

import defaultClasses from './signIn.css';
import classify from 'src/classify';

class SignIn extends Component {
    static propTypes = {
        classes: shape({
            forgotPassword: string,
            form: string,
            modal: string,
            modal_active: string,
            root: string,
            showCreateAccountButton: string,
            signInDivider: string,
            signInError: string,
            signInSection: string
        }),
        isGettingDetails: bool,
        isSigningIn: bool,
        onForgotPassword: func.isRequired,
        setDefaultUsername: func,
        signIn: func,
        signInError: object
    };

    get errorMessage() {
        const { signInError } = this.props;
        const hasError = signInError && Object.keys(signInError).length;

        if (hasError) {
            // Note: we can't access the actual message that comes back from the server
            // without doing some fragile string manipulation. Hardcoded for now.
            return 'The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later.';
        }
    }

    render() {
        const { classes, isGettingDetails, isSigningIn } = this.props;
        const { onSignIn, errorMessage } = this;

        if (isGettingDetails || isSigningIn) {
            return (
                <div className={classes.modal_active}>
                    <LoadingIndicator>Signing In</LoadingIndicator>
                </div>
            );
        } else {
            return (
                <div className={classes.root}>
                    <Form
                        className={classes.form}
                        getApi={this.setFormApi}
                        onSubmit={onSignIn}
                    >
                        <Field label="Email" required={true}>
                            <TextInput
                                autoComplete="email"
                                field="email"
                                validate={isRequired}
                                validateOnBlur
                            />
                        </Field>
                        <Field label="Password" required={true}>
                            <TextInput
                                autoComplete="current-password"
                                field="password"
                                type="password"
                                validate={isRequired}
                                validateOnBlur
                            />
                        </Field>
                        <div className={classes.signInButton}>
                            <Button priority="high" type="submit">
                                Sign In
                            </Button>
                        </div>
                        <div className={classes.signInError}>
                            {errorMessage}
                        </div>
                        <button
                            type="button"
                            className={classes.forgotPassword}
                            onClick={this.handleForgotPassword}
                        >
                            Forgot password?
                        </button>
                    </Form>
                    <div className={classes.signInDivider} />
                    <div className={classes.showCreateAccountButton}>
                        <Button
                            priority="high"
                            onClick={this.showCreateAccountForm}
                        >
                            Create an Account
                        </Button>
                    </div>
                </div>
            );
        }
    }

    handleForgotPassword = () => {
        const username = this.formApi.getValue('email');

        if (this.props.setDefaultUsername) {
            this.props.setDefaultUsername(username);
        }

        this.props.onForgotPassword();
    };

    onSignIn = () => {
        const username = this.formApi.getValue('email');
        const password = this.formApi.getValue('password');

        this.props.signIn({ username, password });
    };

    setFormApi = formApi => {
        this.formApi = formApi;
    };

    showCreateAccountForm = () => {
        const username = this.formApi.getValue('email');

        if (this.props.setDefaultUsername) {
            this.props.setDefaultUsername(username);
        }

        this.props.showCreateAccountForm();
    };
}

export default classify(defaultClasses)(SignIn);
