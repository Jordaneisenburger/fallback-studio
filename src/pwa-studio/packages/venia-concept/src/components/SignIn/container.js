import { connect } from 'src/drivers';
import SignIn from './signIn';
import { signIn } from 'src/actions/user';

const mapStateToProps = ({ user }) => {
    const { isGettingDetails, isSigningIn, signInError } = user;

    return {
        isGettingDetails,
        isSigningIn,
        signInError
    };
};

const mapDispatchToProps = { signIn };

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(SignIn);
