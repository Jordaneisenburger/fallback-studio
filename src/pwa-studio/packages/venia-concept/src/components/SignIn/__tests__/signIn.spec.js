import React from 'react';
import { act } from 'react-test-renderer';
import { Form } from 'informed';
import { createTestInstance } from '@magento/peregrine';

import SignIn from '../signIn';

const props = {
    isGettingDetails: false,
    isSigningIn: false,
    signIn: function() {},
    signInError: {}
};

test('renders correctly', () => {
    const component = createTestInstance(<SignIn {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
});

// TODO: test skipped because of strange exception thrown:
// "TypeError: val.getMockName is not a function"
test.skip('renders the loading indicator if signing in', () => {
    const testProps = {
        ...props,
        isSigningIn: true
    };

    const component = createTestInstance(<SignIn {...testProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('displays an error message if there is a sign in error', () => {
    const testProps = {
        ...props,
        signInError: { message: 'foo ' }
    };

    const component = createTestInstance(<SignIn {...testProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('calls `onSignIn` when sign in button is pressed', () => {
    const signIn = jest.fn();
    const onForgotPassword = jest.fn();

    const { root } = createTestInstance(
        <SignIn signIn={signIn} onForgotPassword={onForgotPassword} />
    );

    act(() => {
        root.findByType(Form).props.onSubmit();
    });

    expect(signIn).toHaveBeenCalledTimes(1);
});
