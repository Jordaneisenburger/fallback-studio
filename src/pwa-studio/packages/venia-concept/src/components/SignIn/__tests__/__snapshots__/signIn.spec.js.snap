// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays an error message if there is a sign in error 1`] = `
<div>
  <form
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div>
      <span>
        <span />
        Email
      </span>
      <span
        style={
          Object {
            "--iconsAfter": 0,
            "--iconsBefore": 0,
          }
        }
      >
        <span>
          <input
            autoComplete="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          />
        </span>
        <span />
        <span />
      </span>
      <p />
    </div>
    <div>
      <span>
        <span />
        Password
      </span>
      <span
        style={
          Object {
            "--iconsAfter": 0,
            "--iconsBefore": 0,
          }
        }
      >
        <span>
          <input
            autoComplete="current-password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            type="password"
            value=""
          />
        </span>
        <span />
        <span />
      </span>
      <p />
    </div>
    <div>
      <button
        type="submit"
      >
        <span>
          Sign In
        </span>
      </button>
    </div>
    <div>
      The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later.
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      Forgot password?
    </button>
  </form>
  <div />
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      <span>
        Create an Account
      </span>
    </button>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <form
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div>
      <span>
        <span />
        Email
      </span>
      <span
        style={
          Object {
            "--iconsAfter": 0,
            "--iconsBefore": 0,
          }
        }
      >
        <span>
          <input
            autoComplete="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          />
        </span>
        <span />
        <span />
      </span>
      <p />
    </div>
    <div>
      <span>
        <span />
        Password
      </span>
      <span
        style={
          Object {
            "--iconsAfter": 0,
            "--iconsBefore": 0,
          }
        }
      >
        <span>
          <input
            autoComplete="current-password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            type="password"
            value=""
          />
        </span>
        <span />
        <span />
      </span>
      <p />
    </div>
    <div>
      <button
        type="submit"
      >
        <span>
          Sign In
        </span>
      </button>
    </div>
    <div />
    <button
      onClick={[Function]}
      type="button"
    >
      Forgot password?
    </button>
  </form>
  <div />
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      <span>
        Create an Account
      </span>
    </button>
  </div>
</div>
`;
